File: food_delivery/src/main/java/com/datn/food_delivery/models/Order.java
package com.datn.food_delivery.models;

import org.springframework.boot.autoconfigure.web.WebProperties;

import java.util.Date;
import java.util.List;

public class Order {
    private Long order_id;
    private double totalAmount;
    private double quantity;
    private Date orderDate;
    private String deliveryAddress;
    private String paymentMethod;
    private String email;
    private Long receiver_id;
    private String Order_Status;

    public Order() {
    }

    public Order(Long order_id, double totalAmount, int quantity, Date orderDate, String deliveryAddress, String paymentMethod, String email, Long receiver_id, String Order_Status) {
        this.order_id = order_id;
        this.totalAmount = totalAmount;
        this.quantity = quantity;
        this.orderDate = orderDate;
        this.deliveryAddress = deliveryAddress;
        this.paymentMethod = paymentMethod;
        this.email = email;
        this.receiver_id = receiver_id;
        this.Order_Status = Order_Status;
    }


    public Long getReceiver_id() {
        return receiver_id;
    }

    public void setReceiver_id(Long receiver_id) {
        this.receiver_id = receiver_id;
    }

    public String getPaymentMethod() {
        return paymentMethod;
    }

    public void setPaymentMethod(String paymentMethod) {
        this.paymentMethod = paymentMethod;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public long getOrder_id() {
        return order_id;
    }

    public void setOrder_id(long order_id) {
        this.order_id = order_id;
    }

    public double getTotalAmount() {
        return totalAmount;
    }

    public void setTotalAmount(double totalAmount) {
        this.totalAmount = totalAmount;
    }

    public double getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }

    public Date getOrderDate() {
        return orderDate;
    }

    public void setOrderDate(Date orderDate) {
        this.orderDate = orderDate;
    }

    public String getDeliveryAddress() {
        return deliveryAddress;
    }

    public void setDeliveryAddress(String deliveryAddress) {
        this.deliveryAddress = deliveryAddress;
    }

    public String getOrder_Status() {
        return Order_Status;
    }

    public void setOrder_Status(String order_Status) {
        Order_Status = order_Status;
    }
}

File: food_delivery_flutter/lib/Model/Order.dart
import 'package:flutter/foundation.dart';

class Order {
  final int order_id;
  final double totalAmount;
  final double quantity;
  final DateTime orderDate;
  final String deliveryAddress;
  final String paymentMethod;
  final String? email;
  final int receiver_id;
  final String order_Status;

  const Order(
      {required this.order_id,
      required this.totalAmount,
      required this.quantity,
      required this.orderDate,
      required this.deliveryAddress,
      required this.paymentMethod,
      required this.email,
      required this.receiver_id,
      required this.order_Status});

  factory Order.fromJson(Map<String, dynamic> json) {
    return Order(
      order_id: json['order_id'],
      totalAmount: json['totalAmount'],
      quantity: json['quantity'],
      orderDate: DateTime.parse(json['orderDate']),
      deliveryAddress: json['deliveryAddress'],
      paymentMethod: json['paymentMethod'],
      email: json['email'],
      receiver_id: json['receiver_id'],
        order_Status: json['order_Status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'order_id': order_id,
      'totalAmount': totalAmount,
      'quantity': quantity,
      'orderDate': orderDate.toIso8601String(),
      'deliveryAddress': deliveryAddress,
      'paymentMethod': paymentMethod,
      'email': email,
      'receiver_id': receiver_id,
      'order_Status': order_Status,
    };
  }

  factory Order.toJson(Map<String, dynamic> json) {
    return Order(
      order_id: json['order_id'],
      totalAmount: json['totalAmount'],
      quantity: json['quantity'],
      orderDate: json['orderDate'],
      deliveryAddress: json['deliveryAddress'],
      paymentMethod: json['paymentMethod'],
      email: json['email'],
      receiver_id: json['receiver_id'],
        order_Status: json['order_Status'],
    );
  }
}
